### ISSUE TITLE
crashing on originLocation

### ISSUE THREAD

**Issue by topinvestor** [mapbox/mapbox-gl-native#13865]
### Steps to reproduce

 1. onMapReady
 2. enableLocationComponent(style)
 3.   crashing on originCoord = new LatLng(originLocation.getLatitude(), originLocation.getLongitude());
                 

### Expected behavior
To get current users' location

### Actual behavior

Crashing

### Configuration

**Android versions:*28* 
**Device models:*Pixel 2* 
**Mapbox SDK versions:*7.0.0*

**Comment by LukasPaczos** [comment]
Hey @topinvestor, thanks for reporting the issue. Unfortunately, I wasn't able to reproduce.

Would you mind sharing more of your setup and point any differences between below?
```
mapView.getMapAsync(new OnMapReadyCallback() {
  @Override
  public void onMapReady(@NonNull final MapboxMap mapboxMap) {
    mapboxMap.setStyle(Style.MAPBOX_STREETS, new Style.OnStyleLoaded() {
      @Override
      public void onStyleLoaded(@NonNull Style style) {
        LocationComponent locationComponent = mapboxMap.getLocationComponent();
        locationComponent.activateLocationComponent(MainActivity.this, style);
        locationComponent.setLocationComponentEnabled(true);
      }
    });
  }
});
```

**Comment by topinvestor** [comment]
This is the scenario:
1) the user opens up the App
2) the App finds the users' location
3) the App gets users' lat/lng to show a pin and callout on the map.

How do you recommend getting the lat/lng once the user location is known?

Thank you,
--Constantine

**Comment by topinvestor** [comment]
Created:

**Comment by LukasPaczos** [comment]
> Obviously last location is still not ready.

This is incorrect. What you are seeing is a [`NullPointerException`](https://docs.oracle.com/javase/7/docs/api/java/lang/NullPointerException.html) which means that the field you are trying to reference, in this case `locationEngine`, doesn't hold a valid reference. You should ensure that you are fetching the right reference with one of the methods I presented above and hold it in your `locationEngine` field. I hope this helps.

In general, this repository should be used for bug reports and feature requests only, feel free to reach out to the [`stackoverflow`](https://stackoverflow.com) community if you need additional assistance.

**Comment by topinvestor** [comment]
On Stackoverflow I post questions all the time and nobody care to answer a basic question.

Here is the problem. I am executing enableLocationComponent and need to know when the location has been found. If I try to get the lat/lng calling getOriginCoord before that it his null and my app crashes.
So basically I need to know how to call getOriginCoord when the Lat/Lng is ready.

Thank you.

---
 @SuppressLint("MissingPermission")
    private void getOriginCoord() {
        Log.i("getOriginCoord", "hello");



        locationEngine.getLastLocation(new LocationEngineCallback<LocationEngineResult>() {
            @Override
            public void onSuccess(LocationEngineResult result) {
   ...

onMapReady->onStyleLoaded->enableLocationComponent

  private void enableLocationComponent(@NonNull Style loadedMapStyle) {
    // Check if permissions are enabled and if not request
        if (PermissionsManager.areLocationPermissionsGranted(this)) {

            // Create and customize the LocationComponent's options
            LocationComponentOptions options = LocationComponentOptions.builder(this)
                    .elevation(5)
                    .accuracyAlpha(.6f)
                    //.accuracyColor(Color.RED)
                    //.foregroundDrawable(R.drawable.android_custom_location_icon)
                    .build();

            // Get an instance of the component
            locationComponent = mapboxMap.getLocationComponent();



            // Activate with options
            locationComponent.activateLocationComponent(this, loadedMapStyle);

            locationEngine = LocationEngineProvider.getBestLocationEngine(this);

            //originLocation = locationComponent.getLastKnownLocation();
            if (locationComponent.getLastKnownLocation() != null) {
                Log.i("getLastKnownLocation","hello getLastKnownLocation" );

            }

            // Enable to make component visible
            locationComponent.setLocationComponentEnabled(true);

    // Set the component's camera mode
            locationComponent.setCameraMode(CameraMode.TRACKING);

    // Set the component's render mode
            locationComponent.setRenderMode(RenderMode.COMPASS);


    // Add the location icon click listener
    locationComponent.addOnLocationClickListener(this);

    // Add the camera tracking listener. Fires if the map camera is manually moved.
    locationComponent.addOnCameraTrackingChangedListener(this);

        } else {
            permissionsManager = new PermissionsManager(this);
            permissionsManager.requestLocationPermissions(this);
        }
    }

**Comment by LukasPaczos** [comment]
You don't have to initialize your own `locationEngine` in the `#enableLocationComponent`, you can do that at any point in time. Then, if you're aiming to receive the first, precise location update, you can use this example.
```
public class SimpleMapActivity extends AppCompatActivity {

  private MapView mapView;
  private final LocationEngineCallbackListener listener = new LocationEngineCallbackListener(this);
  private LocationEngine locationEngine;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_map_simple);
    mapView = findViewById(R.id.mapView);
    mapView.onCreate(savedInstanceState);
    mapView.getMapAsync(new OnMapReadyCallback() {
      @Override
      public void onMapReady(@NonNull MapboxMap mapboxMap) {
        mapboxMap.setStyle(
          Style.MAPBOX_STREETS,
          new Style.OnStyleLoaded() {
            @Override
            public void onStyleLoaded(@NonNull Style style) {
              enableLocationComponent(style);
            }
          }
        );
      }
    });

    locationEngine = LocationEngineProvider.getBestLocationEngine(this);
  }

  private void enableLocationComponent(Style style) {
    // enable component
  }

  void processLocation(@NonNull Location location) {
    // process location
    
    // if you don't want to receive any more updates
    locationEngine.removeLocationUpdates(listener);
  }

  private static final class LocationEngineCallbackListener implements LocationEngineCallback<LocationEngineResult> {
    private final WeakReference<SimpleMapActivity> activityWeakReference;

    LocationEngineCallbackListener(SimpleMapActivity myActivity) {
      this.activityWeakReference = new WeakReference<>(myActivity);
    }

    @Override
    public void onSuccess(LocationEngineResult result) {
      SimpleMapActivity activity = activityWeakReference.get();
      if (activity != null) {
        Location location = result.getLastLocation();
        if (location != null) {
          activity.processLocation(location);
        }
      }
    }

    @Override
    public void onFailure(@NonNull Exception exception) {
      // failure
    }
  }

  @SuppressLint("MissingPermission")
  @Override
  protected void onStart() {
    super.onStart();
    mapView.onStart();
    if (PermissionsManager.areLocationPermissionsGranted(this)) {
      locationEngine.requestLocationUpdates(
        new LocationEngineRequest.Builder(1000)
          .setPriority(LocationEngineRequest.PRIORITY_HIGH_ACCURACY)
          .build(),
        listener,
        Looper.getMainLooper());
    }
  }

  @Override
  protected void onResume() {
    super.onResume();
    mapView.onResume();
  }

  @Override
  protected void onPause() {
    super.onPause();
    mapView.onPause();
  }

  @Override
  protected void onStop() {
    super.onStop();
    mapView.onStop();
    locationEngine.removeLocationUpdates(listener);
  }

  @Override
  public void onLowMemory() {
    super.onLowMemory();
    mapView.onLowMemory();
  }

  @Override
  protected void onDestroy() {
    super.onDestroy();
    mapView.onDestroy();
  }

  @Override
  protected void onSaveInstanceState(Bundle outState) {
    super.onSaveInstanceState(outState);
    mapView.onSaveInstanceState(outState);
  }
}
```

**Comment by topinvestor** [comment]
Hi Łukasz, 

I really appreciate the help. 

Would look at it. 

This is really important because finding My Location is the first thing my App is doing and it should work efficiently and no crashing the App. 

Thank you,
--Constantine

**Comment by topinvestor** [comment]
There is not **LocationEngineCallbackListener** anywhere to be found. Is this a typo?

**Comment by LukasPaczos** [comment]
In case of any other questions please follow just published documentation around the topic: [Android Location Listening](https://docs.mapbox.com/help/tutorials/android-location-listening/).

**Comment by topinvestor** [comment]
Thank you, @LukasPaczos, I would look at it !
